### ISSUE TITLE
GIF is missing `.gif` asset type and `container.data` when loaded Disk Cache

### ISSUE THREAD

**Issue by michaellindahl** [kean/Nuke#682]
I'm running into an issue with the `.gif` asset type. When first downloaded it is successfully matching `imageResponse.container.type == .gif`, but when I load the image again from the Disk Cache it's no longer being returned as a `.gif`. Is when it is being encoded to disk is it losing it's asset type? Where might I look to further diagnose this issue?

This is the GIF in question that I downloaded from giphy.
![zelda](https://user-images.githubusercontent.com/692663/224199269-743ff380-c973-4f5c-ba35-29710ee791d3.gif)

**Comment by kean** [comment]
Hey, you you please share your pipeline configuration?

**Comment by michaellindahl** [comment]
This occurs when I use the `DataCache` rather than the default `URLCache`:
```
        let configuration = ImagePipeline.Configuration.withDataCache()
        ImagePipeline.shared = ImagePipeline(configuration: configuration)
```

**Comment by kean** [comment]
I can reproduce it, but only with [.storeEncodedImages](https://kean-docs.github.io/nuke/documentation/nuke/imagepipeline/datacachepolicy/storeencodedimages) cache policy. It's a defect. It doesn't currently support GIFs. It uses `ImageEncoders.Default` for encoding.

**Comment by michaellindahl** [comment]
What is your stance on having Nuke fully support GIFs using ImageIO? 

I think previously Nuke was embedding GIFU. But I think we could use `ImageIO` without adding too much complexity. We could use `CGAnimateImageDataWithBlock` to animate the data in `container.data`. And then I we could apply the transformation pipelines on each frame with `CGImageSourceGetCount`/ `CGImageSourceCreateImageAtIndex` / etc.

**Comment by kean** [comment]
Yes, it was previously using [Gifu](https://github.com/kaishin/Gifu) internally, but now the recommendation is to use it, or any other framework that supports GIF, directly. One of the main Nuke advantages is how small the code size is compared to the competitive frameworks, and I would like to keep it this way.

**Comment by kean** [comment]
Fixed in [Nuke 12.1](https://github.com/kean/Nuke/releases/tag/12.1.0).
